From bcbc834dba2feab9912b2b8243cbf8225c8f425a Mon Sep 17 00:00:00 2001
From: Dusk Banks <me@bb010g.com>
Date: Sun, 2 Feb 2025 06:37:34 -0800
Subject: [PATCH 1/2] CMake: non-internal RPATH cache variables

SPDX-FileCopyrightText: Dusk Banks <me@bb010g.com>
SPDX-License-Identifier: Apache-2.0 OR MIT
---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f27858f..d565380 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,14 +11,14 @@ project(luvi ${projects})
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
 # use, i.e. don't skip the full RPATH for the build tree
-SET(CMAKE_SKIP_BUILD_RPATH FALSE CACHE INTERNAL "")
+SET(CMAKE_SKIP_BUILD_RPATH FALSE CACHE BOOL "")
 # when building, don't use the install RPATH already
 # (but later on when installing)
-SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE CACHE INTERNAL "")
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib" CACHE INTERNAL "")
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE CACHE BOOL "")
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib" CACHE STRING "")
 # add the automatically determined parts of the RPATH
 # which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE INTERNAL "")
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE BOOL "")
 # the RPATH to be used when installing, but only if it's not a system directory
 LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
 IF("${isSystemDir}" STREQUAL "-1")
-- 
2.47.0

